src/main/webapp/search_area.js

```diff
@@ -16,32 +16,52 @@ document.addEventListener("DOMContentLoaded", async function() {
   // This checks that requester User has valid credentials to edit/delete/view ALL organizations. (by checking userID)
   let isMaintainer = false;
   let forOrganizationsPage = false;
-
+  let forEventsPage = false;
   if (document.getElementById('search-area')) {
-    const mainSearchArea = new SearchArea(document.getElementById('search-area'), isMaintainer, forOrganizationsPage);
+    const mainSearchArea = new SearchArea(document.getElementById('search-area'), isMaintainer, forOrganizationsPage, forEventsPage);
     mainSearchArea.handleOrganizations();
   }
-  isMaintainer = true;
   if (document.getElementById('all-organizations')) {
     forOrganizationsPage = true;
-    const organizationSearchArea = new SearchArea(document.getElementById('all-organizations'), isMaintainer, forOrganizationsPage);
+    const organizationSearchArea = new SearchArea(document.getElementById('all-organizations'), isMaintainer, forOrganizationsPage, forEventsPage);
     organizationSearchArea.handleOrganizations();
   }
+  if (document.getElementById('events')) {
+      forEventsPage = true;
+      const eventsSearchArea = new SearchArea(document.getElementById('events'), isMaintainer, forOrganizationsPage, forEventsPage);  }
 });
 
 class SearchArea {
-  constructor(searchAreaElement, isMaintainer, forOrganizationsPage) {
-    this.searchArea = searchAreaElement;
-    this.organizationSearchArea = document.createElement("div");
+  constructor(searchAreaElement, isMaintainer, forOrganizationsPage, forEventsPage) {
+    this.searchAreaContainer = searchAreaElement;
+    this.searchArea = document.createElement("div");
     this.isMaintainer = isMaintainer;
+    // TODO(): Add isModerator check after moderator PR is pulled in
+    this.isModerator = true;
     this.forOrganizationsPage = forOrganizationsPage;
+    this.forEventsPage = forEventsPage;
     this.filterParams = new URLSearchParams();
     this.organizationObjectsList = [];
+    this.eventObjectsList = [];
```

Comment:
Since this code never operates in a situation where organizationObjectsList AND eventObjectsList need to have values, what if we generalized organizationObjectsList/eventObjectsList to be something like "objectsList", and then the events and organizations are both filled in there? This way, we would only need to change lines 145 and 174 to use different classes for the 2 situations, and different servlets, respectively.

We would also be able to do the same thing with organizationListArea / eventListArea: we could reference the list area by a single name in the code at places like line 132, 162, etc. but just assign that list area in the constructor to either be an event list area or org list area.

If we can generalize the repeated objects into the same ones for both cases, it will be very easy to connect this to the backend

Reply:
okay!